bugsnag/src/main/java/com/bugsnag/Bugsnag.java

```diff
@@ -601,6 +611,45 @@ public void close() {
         ExceptionHandler.disable(this);
     }
 
+    // Thread metadata
+
+    /**
+     * Add a key value pair to a metadata tab just for this thread.
+     *
+     * @param tabName the name of the tab to add the key value pair to
+     * @param key     the key of the metadata to add
+     * @param value   the metadata value to add
+     */
+    public static void addThreadMetaData(String tabName, String key, Object value) {
```

Comment:
do we want to think about providing instance methods for this (maybe in addition to these existing methods)?

Reply:
Why? These seem like static methods all day long to me.

Reply:
Most of the methods in our public Java API are instance methods. I felt it was worth raising the point to check that this feels consistent, and also to check that we're not making it static for all use-cases simply because we need to call this method in that way from within the notifier.

Reply:
My preference is for methods that don't require an instance should be made static - but maybe that isnt a very java-ey thing to say. You can call a static method on an instance - but that is another can of worms https://www.quora.com/Why-cant-we-call-static-methods-using-the-object-in-Java